cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/AppMetadataStoreTest.java

```diff
@@ -136,6 +139,46 @@ public Long apply(RunId input) {
     Assert.assertEquals((countingTicker.getNumProcessed() - (2 * numBatches)) / maxScanTimeMillis, numBatches);
   }
 
+  @Test
+  public void testgetRuns() throws Exception {
+    DatasetId storeTable = NamespaceId.DEFAULT.dataset("testgetRuns");
+    datasetFramework.addInstance(Table.class.getName(), storeTable, DatasetProperties.EMPTY);
+
+    Table table = datasetFramework.getDataset(storeTable, ImmutableMap.<String, String>of(), null);
+    Assert.assertNotNull(table);
+    AppMetadataStore metadataStoreDataset = new AppMetadataStore(table, cConf);
+
+    // Add some run records
+    Set<String> expected = new TreeSet<>();
+
+    Set<ProgramRunId> programRunIdSet = new HashSet<>();
+    for (int i = 0; i < 100; ++i) {
+      ApplicationId application = NamespaceId.DEFAULT.app("app" + i);
+      ProgramId program = application.program(ProgramType.values()[i % ProgramType.values().length],
+                                              "program" + i);
+      RunId runId = RunIds.generate((i + 1) * 10000);
+      expected.add(runId.toString());
+
+      ProgramRunId programRunId = new ProgramRunId(program.getNamespace(), program.getApplication(),
+                                                   program.getType(), program.getProgram(), runId.toString());
+      programRunIdSet.add(programRunId);
+
+      // Start the program and stop it
+      metadataStoreDataset.recordProgramStart(program, runId.getId(), RunIds.getTime(runId, TimeUnit.SECONDS),
+                                              null, null, null);
+      metadataStoreDataset.recordProgramStop(program, runId.getId(), RunIds.getTime(runId, TimeUnit.SECONDS),
+                                             ProgramRunStatus.values()[i % ProgramRunStatus.values().length], null);
+    }
+
+    Map<ProgramRunId, RunRecordMeta> runMap =  metadataStoreDataset.getRuns(programRunIdSet);
```

Comment:
It would be good to add another assertion for fetching `RunRecordMeta` fewer run ids, say 50% of all run ids in the table.